index.js

```diff
@@ -442,6 +447,18 @@ class BlindPeer extends ReadyResource {
     stream.on('close', () => core.close().catch(safetyCatch))
   }
 
+  async _announceCores() {
+    for await (const record of this.db.createAnnouncingCoresStream()) {
+      if (this.closing) return
+      await this._announceCore(record.key)
+      if (this.closing) return
+      await new Promise((resolve) => setTimeout(resolve, this.announcingInterval))
+      if (this.closing) return
+    }
+
+    this.emit('announce-cores')
```

Comment:
Can you rename ton 'announced-cores'? Then it's clear from the name that it's emitted only after they're all announced ('announce-cores' could also mean it begins announcing)